src/sentry/search/eap/spans.py

```diff
@@ -0,0 +1,253 @@
+
+    def resolve_query(self, querystring: str) -> TraceItemFilter | None:
+        """Given a query string in the public search syntax eg. `span.description:foo` construct the TraceItemFilter"""
+        try:
+            parsed_terms = event_search.parse_search_query(
+                querystring,
+                params=self.params.filter_params,
+                get_field_type=self.get_field_type,
+            )
+        except ParseError as e:
+            if e.expr is not None:
+                raise InvalidSearchQuery(f"Parse error: {e.expr.name} (column {e.column():d})")
+            else:
+                raise InvalidSearchQuery(f"Parse error for: {querystring}")
+
+        if any(
+            isinstance(term, event_search.ParenExpression)
+            or event_search.SearchBoolean.is_operator(term)
+            for term in parsed_terms
+        ):
+            return self._resolve_boolean_conditions(parsed_terms)
+        else:
+            return self._resolve_terms(parsed_terms)
+
+    def _resolve_boolean_conditions(
+        self, terms: event_filter.ParsedTerms
+    ) -> TraceItemFilter | None:
+        if len(terms) == 1:
+            if isinstance(terms[0], event_search.ParenExpression):
+                return self._resolve_boolean_conditions(terms[0].children)
+            elif isinstance(terms[0], event_search.SearchFilter):
+                return self._resolve_terms([cast(event_search.SearchFilter, terms[0])])
+            else:
+                raise NotImplementedError("Haven't handled all the search expressions yet")
+
+        # Filter out any ANDs since we can assume anything without an OR is an AND. Also do some
+        # basic sanitization of the query: can't have two operators next to each other, and can't
+        # start or end a query with an operator.
+        previous_term: event_filter.ParsedTerm | None = None
+        new_terms = []
+        term: event_filter.ParsedTerm | None = None
+        for term in terms:
+            if previous_term:
+                if event_search.SearchBoolean.is_operator(
+                    previous_term
+                ) and event_search.SearchBoolean.is_operator(term):
+                    raise InvalidSearchQuery(
+                        f"Missing condition in between two condition operators: '{previous_term} {term}'"
+                    )
+            else:
+                if event_search.SearchBoolean.is_operator(term):
+                    raise InvalidSearchQuery(
+                        f"Condition is missing on the left side of '{term}' operator"
+                    )
+
+            if term != event_search.SearchBoolean.BOOLEAN_AND:
+                new_terms.append(term)
+
+            previous_term = term
+
+        if term is not None and event_search.SearchBoolean.is_operator(term):
+            raise InvalidSearchQuery(f"Condition is missing on the right side of '{term}' operator")
```

Comment:
What happens if `terms` is an empty sequence? Think the empty search string.

Reply:
Good callout, shouldn't get here cause this is the boolean search codepath, but does fail in `_resolve_terms`